lib/sync/pull-request.js

```diff
@@ -1,57 +1,97 @@
- * @param {any} cursor - TBD
+ * @param {RepositoryObject} repository - TBD
+ * @param {string|number} cursor - TBD
  * @param {any} perPage - TBD
  * @returns {any} TBD
  */
-exports.getPullRequests = async (github, repository, cursor, perPage) => {
+async function getPullRequests(github, repository, cursor, perPage) {
+  /** @type {number} */
+  let status;
+  /** @type {{link: string}} */
+  let headers;
+  /** @type {import('@octokit/rest').Octokit.PullsListResponse} */
   let edges;
+  if (!cursor) {
+    cursor = 1;
+  } else {
+    cursor = Number(cursor);
+  }
   const vars = {
     owner: repository.owner.login,
     repo: repository.name,
     per_page: perPage,
-    cursor,
+    page: cursor,
   };
 
   let asyncTags = [];
   await statsd.asyncTimer(
     async () => {
-      try {
-        ({ edges } = (await github.graphql(getPullRequestQuery, vars)).repository.pullRequests);
-        asyncTags.push('status:200');
-      } catch (error) {
-        logger.error('GraphQL Query Error for Sync Pull Request', error);
-        if (error.status) {
-          asyncTags.push(`status:${error.status}`);
-        } else if (error.httpError && error.httpError.status) {
-          asyncTags.push(`status:${error.httpError.status}`);
-        } else {
-          asyncTags.push('status:exception');
-        }
-        throw error;
-      }
+      ({ data: edges, status, headers } = (await github.pulls.list({
+        ...vars,
+        state: 'all',
+        sort: 'created',
+        direction: 'desc',
+        // Retry up to 6 times pausing for 10s, for *very* large repos we need to wait a while for the result to succeed in dotcom
+        request: {
+          retries: 6,
+          retryAfter: 10,
+        },
+      })));
+      asyncTags.push(`status:${status}`);
     },
-    'graphql.sync_pull_request',
+    'rest.sync_pull_request',
     1,
     asyncTags,
   )();
 
-  let pullRequests = await Promise.all(edges.map(async ({ node: pull }) => {
-    // Count of comments causes graphql to time out, fetch it from the regular API
-    const pr = await github.pulls.get({
-      owner: repository.owner.login,
-      repo: repository.name,
-      pull_number: pull.number,
-    });
-    pull.comments = { totalCount: pr.data.comments };
-    const { data } = transformPullRequest({ pull_request: pull, repository }, pull.author);
+  // Force us to go to a non-existant page if we're past the max number of pages
+  const nextPage = getNextPage(headers) || cursor + 1;
+
+  // Attach the "cursor" (next page number) to each edge, because the function that uses this data
+  // fetches the cursor from one of the edges instead of letting us return it explicitly.
+  edges.forEach((edge) => {
+    edge.cursor = nextPage;
+  });
+
+  let pullRequests = await Promise.all(edges.map(async (pull) => {
+    const { data } = await transformPullRequest({ pullRequest: pull, repository }, pull.user, github);
     return data && data.pullRequests[0];
```

Comment:
This returns a PR? It reads like a boolean to me 😆

Reply:
oh, no, you're right. lolol. It returns a boolean. WHY IS THIS CODE SO CONFUSING